File: Modules/Common/FMS/SIE.Web.FMS/FileManage/Commands/AddEmployeeCommand.cs
using SIE.Domain;
using SIE.FMS;
using SIE.Web.Command;
using System.Collections.Generic;

namespace SIE.Web.FMS.FileManage.Commands
{
    /// <summary>
    /// 添加员工命令
    /// </summary>
    [JsCommand("SIE.Web.FMS.FileManage.Commands.AddEmployeeCommand")]
    public class AddEmployeeCommand : ViewCommand
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">scope</param>
        /// <returns>返回结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var savedData = new EntityList<UserInFileUserGroup>();
            var userInFileUserGroupList = args.Data.ToJsonObject<List<UserInFileUserGroup>>();
            Check.NotNullOrEmpty(userInFileUserGroupList, nameof(userInFileUserGroupList));
            foreach (var item in userInFileUserGroupList)
            {
                var userInFileUserGroup = new UserInFileUserGroup();
                userInFileUserGroup.FileUserGroupId = item.FileUserGroupId;
                userInFileUserGroup.EmployeeId = item.EmployeeId;
                savedData.Add(userInFileUserGroup);
            }
            RF.Save(savedData);
            return true;
        }
    }
}

File: Modules/Common/FMS/SIE.Web.FMS/FileManage/Commands/AddEmployeeCommand.js
SIE.defineCommand('SIE.Web.FMS.FileManage.Commands.AddEmployeeCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'EmployeeId', targetClassName: 'SIE.Resources.Employee' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        /// <summary>
        /// 保存选择的操作列表。
        /// </summary>
        var me = this;
        /* post数据结构*/
        var indata = {};
        /* post数据结构*/
        var selections = this._targetSelectItems.items;
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var employeeId = item.getId();
                if (me._sourceViewSelectItems.indexOf(employeeId) === -1) {
                    var userInFileUserGroup = { FileUserGroupId: me._sourceId, EmployeeId: employeeId };
                    operationDatas.push(userInFileUserGroup);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();
                    me._ownerView.loadChildData(true);
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.L10N());
        }
    }
});
